test/test_optim.py

```diff
@@ -1547,5 +1547,84 @@ def test_cosine_then_cyclic(self):
 
         self.assertLessEqual(last_lr, max_lr)
 
+class TestStochasticRoundingOptim(TestCase):
```

Comment:
The tests might need to be skipped, if no CUDA device is available via `@unittest.skipIf(not torch.cuda.is_available(), 'No CUDA')`